### PR TITLE
Create API For Edit Gym and test it 

### PR DESCRIPTION
**Pull request by AFahgan**

### CODE REVIEW

**Review by AhmedQeshta — APPROVED** [review]

**Comment by AhmedQeshta** [comment]
Fix Conflicting files #93 

@AFahgan

**Comment by AFahgan** [comment]
> Fix Conflicting files #93
> 
> @AFahgan



> Fix Conflicting files #93
> 
> @AFahgan

I'll do them after pushing the tests.

**Review by rawaneyad — APPROVED** [review]
Good Job Ayman 👍🏼

File: server/utils/validation/gymEditSchema.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import Joi from 'joi';
+
+const gymRegisterSchema = Joi.object({
+  gymName: Joi.string().required(),
+  logo: Joi.string().required(),
+  email: Joi.string()
+    .max(250)
+    .min(4)
+    .email({ minDomainSegments: 2, tlds: { allow: ['com', 'net'] } })
+    .required(),
```
Comment (LinaYahya) [inline server/utils/validation/gymEditSchema.ts:10]: we will not send email on edit profile, nor password, an we will send gym ID in the request body

File: server/controllers/gym/editGym.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+export default async function editGymData(req: Request, res: Response, next: NextFunction) {
+  try {
+    await gymEditSchema.validateAsync(req.body);
+    const { id } = req.params;
+    const gym = await Gym.findByPk(id);
+    if (!gym) {
+      throw new CustomError('Gym not found', 404);
+    }
+    await gym.update(req.body);
```
Comment (LinaYahya) [inline server/controllers/gym/editGym.ts:14]: here id  I have a gym token, I'm able to update any gym data,
So we need first in the route itself to use `checkAuth('gym')`, that will put gym id in the request object,
then we have to check id in the request from checkAuth with id in the request body, If they are equivalent then user is authorized to update the data

Comment (LinaYahya) [inline server/controllers/gym/editGym.ts:14]: What if user changed the logo, then we need to upload image to `aws`,
So check before `gym.logo === req.body.logo`, if those differs then we need to upload image first

File: server/controllers/gym/editGym.ts [hunk 3]
```diff
@@ -0,0 +1,20 @@
+import { NextFunction, Response, Request } from 'express';
+// import Sequelize from 'sequelize';
+import Gym from '../../database/models/gyms';
```
Comment (LinaYahya) [inline server/controllers/gym/editGym.ts:3]:
```suggestion
import {Gym} from '../../database/models';
```

**Review by LinaYahya — CHANGES_REQUESTED** [review]

File: server/controllers/gym/editGym.ts [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+import { CustomError, gymEditSchema } from '../../utils';
+
+export default async function editGymData(req: Request, res: Response, next: NextFunction) {
```
Comment (AhmedQeshta) [inline server/controllers/gym/editGym.ts:6]: Here You need Check if the gym owner Requested a new logo > remove the old logo from **AWS** then upload a new image.

File: server/utils/validation/gymEditSchema.ts [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+  password: Joi.string().max(25).min(8).required(),
```
Comment (AhmedQeshta) [inline server/utils/validation/gymEditSchema.ts:11]:
```suggestion

```
we will not send an email on edit profile, or password, and we will send gym ID in the request body

**Review by AhmedQeshta — CHANGES_REQUESTED** [review]

**Pull request closed without merging**